doc/source/library_description/packaging.rst

```diff
@@ -2,73 +2,229 @@
-* pkg_resources-style namespace packages
 
 Required Files
 --------------
 
-* README.rst file: Describes the purpose of the package.
+* ``README.rst`` file: Describes the purpose of the package.
   *The format of this file must be reStructuredText.*
 
-* LICENSE file: Specifies copyrights and required authorization.
+* ``LICENSE`` file: Specifies copyrights and required authorization.
 
-* setup.py file: Provides package information.
-  The presence of this file indicate that the package was likely created using ``disutils``,
-  which is the Python standard for building and distributing a Python package.
+* ``pyproject.toml`` file: Provides package metadata and defines how the package
+  is built. There are different build backends available, such as `setuptools`_,
+  `poetry`_, and `flit`_.
 
+* ``src/ansys/product/library/__init__.py`` file: Usually contains the
+  version of the package in a variable named ``__version__``. The value of this
+  variable can be parsed from the ``pyproject.toml`` file so that the version 
+  is only specified in one location.
 
-Setup File
-----------
-The `setup.py`_ file is the build script for ``setuptools``. It exposes dynamic metadata and contains
-package information, such as a description, author, and version.
-In this file, the ``setuptools`` module is used to configure the metadata (as opposed to ``distutils``).
 
-.. code:: python
+Additional Directories
+----------------------
+
+The following directories may be specified at the same level as the ``src/`` one:
+
+* ``tests/``: Contains all unit tests for the package. It is
+  likely that these tests take advantage of the `pytest`_ framework.
+
+* ``doc/``: Contain all documentation files and examples on
+  how to use the package.
+
+
+Project File and Build System
+------------------------------
+
+The ``pyproject.toml`` file is the standardized build configuration file for Python
+projects. It must contain at least a ``[build-system]`` section, which determines
+how the project is built. Some commonly used packaging tools are `setuptools`_,
+`poetry`_, and `flit`_. All three of these packaging tools are currently supported by
+the `pyansys template`_.
+
+
+Flit
+^^^^
+
+Flit is a modern and lightweight build system that requires developers
+to manage virtual environments on their own. Developers must:
+
+* Create a virtual environment and activate it.
+* Install the package in editable mode.
+
+Flit is the default tool for creating a new project when using the
+`pyansys template`_.
+
+The ``[project]`` section specifies the project's metadata and required
+dependencies. For more information, see the `flit pyproject.toml
+guidelines`_.
+
 
-  import setuptools
-  setuptools.setup(...)
+Poetry
+^^^^^^
+
+Poetry is known because of its strong dependency pinning thanks to the
+``poetry.lock`` file. When installing a package, poetry creates a virtual
+environment, thus ensuring an isolated package development environment.
```

Comment:
```suggestion
 Because of its ``poetry.lock`` file, Poetry provides strong dependency pinning. When
 installing a package, poetry creates a virtual environment, thus ensuring an isolated
 package development environment.
```